### ISSUE TITLE
Upgrade to tomcat 7.0.59

### ISSUE THREAD

**Issue by markgarg** [jsimone/webapp-runner#47]
Problem with server startup when using webapp-runner version 7.0.57.2.
I have created a basic Maven Web Application using Eclipse and wanted to deploy it to Heroku. Config:
JDK - 1.8 / 1.7 (tried with both)
Servlet container - 3.0.1 / 3.1
webapp-runner version - 7.0.57.1/7.0.57.2

[Bug status from Apache](https://bz.apache.org/bugzilla/show_bug.cgi?id=57173) shows it's resolved from 7.0.58 onward.

I'm wondering if there's something wrong with what I'm doing or if it's a problem with the Tomcat version itself ?
```
Jul 10, 2015 12:34:51 PM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-nio-8080"]
Jul 10, 2015 12:34:51 PM org.apache.tomcat.util.net.NioSelectorPool getSharedSelector
INFO: Using a shared selector for servlet write/read
Jul 10, 2015 12:34:51 PM org.apache.catalina.core.StandardService startInternal
INFO: Starting service Tomcat
Jul 10, 2015 12:34:51 PM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet Engine: Apache Tomcat/7.0.57
Jul 10, 2015 12:34:52 PM org.apache.catalina.startup.ContextConfig getDefaultWebXmlFragment
INFO: No global web.xml found
Jul 10, 2015 12:34:54 PM org.apache.catalina.startup.ContextConfig processAnnotationsJar
SEVERE: Unable to process Jar entry [org/apache/batik/dom/svg/SVGOMAnimatedPathData$AnimSVGPathSegList.class] from Jar [jar:jndi:/localhost/WEB-INF/lib/batik-svg-dom-1.7.jar!/] for annotations
java.io.EOFException
	at org.apache.tomcat.util.bcel.classfile.FastDataInputStream.readUnsignedShort(FastDataInputStream.java:120)
	at org.apache.tomcat.util.bcel.classfile.Utility.swallowFieldOrMethod(Utility.java:75)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.readMethods(ClassParser.java:235)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.parse(ClassParser.java:92)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2071)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:1947)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:1913)
	at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:1898)
	at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1330)
	at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:889)
	at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:386)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:117)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:90)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5380)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1575)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1565)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Jul 10, 2015 12:34:54 PM org.apache.catalina.startup.ContextConfig processAnnotationsJar
SEVERE: Unable to process Jar entry [org/apache/batik/dom/svg/SVGOMStyleElement$DOMCharacterDataModifiedListener.class] from Jar [jar:jndi:/localhost/WEB-INF/lib/batik-svg-dom-1.7.jar!/] for annotations
java.io.EOFException
	at org.apache.tomcat.util.bcel.classfile.FastDataInputStream.readInt(FastDataInputStream.java:145)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.readID(ClassParser.java:200)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.parse(ClassParser.java:78)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2071)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:1947)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:1913)
	at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:1898)
	at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1330)
	at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:889)
	at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:386)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:117)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:90)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5380)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1575)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1565)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Jul 10, 2015 12:34:54 PM org.apache.catalina.startup.ContextConfig processAnnotationsJar
SEVERE: Unable to process Jar entry [org/apache/xalan/xsltc/compiler/Parser.class] from Jar [jar:jndi:/localhost/WEB-INF/lib/xalan-2.7.1.jar!/] for annotations
java.io.EOFException
	at org.apache.tomcat.util.bcel.classfile.FastDataInputStream.readUnsignedShort(FastDataInputStream.java:120)
	at org.apache.tomcat.util.bcel.classfile.Utility.swallowFieldOrMethod(Utility.java:75)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.readMethods(ClassParser.java:235)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.parse(ClassParser.java:92)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2071)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:1947)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:1913)
	at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:1898)
	at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1330)
	at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:889)
	at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:386)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:117)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:90)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5380)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1575)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1565)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Jul 10, 2015 12:34:55 PM org.apache.catalina.startup.ContextConfig processAnnotationsJar
SEVERE: Unable to process Jar entry [org/apache/poi/hssf/converter/ExcelToFoConverter.class] from Jar [jar:jndi:/localhost/WEB-INF/lib/poi-scratchpad-3.8.jar!/] for annotations
java.io.EOFException
	at org.apache.tomcat.util.bcel.classfile.FastDataInputStream.readUnsignedShort(FastDataInputStream.java:120)
	at org.apache.tomcat.util.bcel.classfile.Utility.swallowFieldOrMethod(Utility.java:75)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.readMethods(ClassParser.java:235)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.parse(ClassParser.java:92)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2071)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:1947)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:1913)
	at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:1898)
	at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1330)
	at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:889)
	at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:386)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:117)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:90)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5380)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1575)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1565)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Jul 10, 2015 12:34:55 PM org.apache.catalina.startup.ContextConfig processAnnotationsJar
SEVERE: Unable to process Jar entry [org/apache/commons/lang/StringUtils.class] from Jar [jar:jndi:/localhost/WEB-INF/lib/commons-lang-2.4.jar!/] for annotations
java.io.EOFException
	at org.apache.tomcat.util.bcel.classfile.FastDataInputStream.readUnsignedShort(FastDataInputStream.java:120)
	at org.apache.tomcat.util.bcel.classfile.Utility.swallowFieldOrMethod(Utility.java:75)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.readMethods(ClassParser.java:235)
	at org.apache.tomcat.util.bcel.classfile.ClassParser.parse(ClassParser.java:92)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsStream(ContextConfig.java:2071)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsJar(ContextConfig.java:1947)
	at org.apache.catalina.startup.ContextConfig.processAnnotationsUrl(ContextConfig.java:1913)
	at org.apache.catalina.startup.ContextConfig.processAnnotations(ContextConfig.java:1898)
	at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1330)
	at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:889)
	at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:386)
	at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:117)
	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:90)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5380)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1575)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1565)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.lang.Thread.run(Unknown Source)

Jul 10, 2015 12:34:58 PM org.apache.catalina.util.SessionIdGenerator createSecureRandom
INFO: Creation of SecureRandom instance for session ID generation using [SHA1PRNG] took [150] milliseconds.
Jul 10, 2015 12:34:58 PM org.apache.coyote.AbstractProtocol start
INFO: Starting ProtocolHandler ["http-nio-8080"]
```

**Comment by jkutner** [comment]
I looks like there are compatibility issues between 7.0.59+ and one of our dependencies, `ru.zinin/tomcat-redis-session`. So this may take some time.

Have you tried 8.0.18.0-M1?

**Comment by markgarg** [comment]
Ah, thanks for the quick reply !
I've just checked in Maven Central and found that "8.0.18.0-M1" is published. (headsmack for assuming pull request wouldn't have been published).

Got it and tested. __8.0.18.0-M1 works__ !

**Comment by jkutner** [comment]
That's good news. We're looking to move all of our system to 8. I've also just released 8.0.23.0, which should be available in the next hour or so.
